### PR TITLE
Zeitwerkify

### PR DESCRIPTION
**Pull request by solnic**

### CODE REVIEW

File: dry-core.gemspec [hunk 1]
```diff
@@ -30,6 +30,7 @@ Gem::Specification.new do |spec|
 
   # to update dependencies edit project.yml
   spec.add_runtime_dependency "concurrent-ruby", "~> 1.0"
+  spec.add_runtime_dependency "zeitwerk", "~> 1.4"
```
Comment (flash-gordon) [inline dry-core.gemspec:33]: this must be here https://github.com/dry-rb/dry-core/blob/250ad54dfa6bc1277c4af64bdccbe045d96d04bd/project.yml#L11

File: lib/dry/core/class_attributes.rb [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-require "dry/core/constants"
-require "dry/core/errors"
-
 module Dry
   module Core
     # Internal support module for class-level settings
     #
     # @api public
     module ClassAttributes
-      include Constants
```
Comment (flash-gordon) [inline lib/dry/core/class_attributes.rb:12]: But this should have broken tests :)

File: dry-core.gemspec (hunk 1 above)
Comment (solnic) [reply dry-core.gemspec:33]: funny that I even I forgot 😅 thanks! I'll be fixing all of that in main branches.

File: lib/dry/core/class_attributes.rb (hunk 2 above)
Comment (solnic) [reply lib/dry/core/class_attributes.rb:12]: @flash-gordon I included `Constants` in `Dry::Core`

Comment (flash-gordon) [reply lib/dry/core/class_attributes.rb:12]: Ah OK

**Pull request merged**